src/switch_core_media.c

```diff
@@ -2088,17 +2088,30 @@ SWITCH_DECLARE(int) switch_core_session_check_incoming_crypto(switch_core_sessio
 		} else {
 			const char *a = switch_stristr("AE", engine->ssec[engine->crypto_type].remote_crypto_key);
 			const char *b = switch_stristr("AE", crypto);
+			switch_bool_t skip_local = 0;
 
-			if (sdp_type == SDP_TYPE_REQUEST) {
-				if (!vval) {
-					switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_DEBUG, "Unsupported Crypto [%s]\n", crypto);
-					goto end;
-				}
-				switch_channel_set_variable(session->channel, varname, vval);
+			/* Skip building new local key if only remote mki is added or removed */
+			if (switch_channel_var_true(session->channel, "skip_empty_rtp_secure_media_mki") 
+				&& (!strcmp((switch_core_session_sprintf(smh->session, "%s|1:1", crypto)), engine->ssec[engine->crypto_type].remote_crypto_key) 
```

Comment:
It shouldn't be static to 1:1. Use regex instead to match the key e.g. %d+:%d+

Reply:
Using regex doesn't seem to work with `switch_core_session_sprintf`. Used `strstr` instead